apps/organisations/middleware.py

```diff
@@ -0,0 +1,20 @@
+from django.shortcuts import get_object_or_404
+from django.utils.deprecation import MiddlewareMixin
+
+from . import clear_organisation
+from . import set_organisation
+from .models import Organisation
+
+
+class OrganisationMiddleware(MiddlewareMixin):
```

Comment:
Should we make this more explicit and update our urls instead of using the middleware? It would make the code easier to understand. What do you think?

Reply:
Yep definitely, but there I was still hanging. So I'd say we do this in a follow up PR?

Reply:
sounds good